src/coreclr/jit/codegenarm.cpp

```diff
@@ -460,7 +460,7 @@ void CodeGen::genLclHeap(GenTree* tree)
         }
 
         // regCnt will be the total number of bytes to locAlloc
-        genSetRegToIcon(regCnt, amount, ((int)amount == amount) ? TYP_INT : TYP_LONG);
+        genSetRegToIcon(regCnt, amount, ((int)amount >= 0) ? TYP_INT : TYP_LONG);
```

Comment:
this does not look right, a narrowing cast is implementation-defined in C++.
also, it looks like we don't support TYP_LONG in https://github.com/dotnet/runtime/blob/eee0ed206b9277f3ade51af3ceb3aadb6ff609c0/src/coreclr/jit/codegenarm.cpp#L183
that we are calling here, maybe use `ClrSafeInt ` and cast to int always? cc @echesakovMSFT

Reply:
Yes, my bad, thanks. But it seems that original code is incorrect too

Reply:
I don't think this is relevant to `CodeGen::instGen_Set_Reg_To_Reloc_Imm` issue in cross-bitness scenarious.

The change is for `localloc` (i.e. .NET way of `alloca`). 

According to ECMA-335 

*The `localloc` instruction allocates size (type `native unsigned int` or `U4`) bytes from the local
dynamic memory pool ...*

I agree with Sergey that we should always use `TYP_INT` here.
```suggestion
        genSetRegToIcon(regCnt, amount, TYP_INT);
```

Reply:
Fixed, thanks